openstack/nova/shellinabox/revproxy.lua

```diff
@@ -0,0 +1,226 @@
+local _M = {}
+local cjson = require "cjson"
+local config = require "config"
+local resty_sha256 = require "resty.sha256"
+local str = require "resty.string"
+local memcached = require "resty.memcached"
+local mysql = require "resty.mysql"
+local mlcache_loaded, mlcache = pcall(require, "resty.mlcache")
+local x = "%x"
+local t = { x:rep(8), x:rep(4), x:rep(4), x:rep(4), x:rep(12) }
```

Comment:
What that do? Is that string-formatting for numbers?

Reply:
Well, yes and no. It is the same "code", but for matching [patterns](https://www.lua.org/manual/5.3/manual.html#6.4.1). They are similar to regular expressions, but less powerful/complex.
The alternative would be hard-coding the "%x%x%x..-%x%x"... not sure if that is easier to read. I was fearing, I might miss a `%x` somewhere.

Reply:
Ah. It means "repeat". Thanks.